doc/device_client.md

```diff
@@ -0,0 +1,32 @@
+# Azure IoT Device Client Library
+
+The Azure IoT Device Client Library provides an efficient and well tested API for connecting and using devices with the Azure IoT Hub.
+
+## client.setOptions(options, [callback])
```

Comment:
(fyi, I'm going to be asking naive questions like a customer with no knowledge of the SDK)-- in what circumstance would I add a callback to the setOptions call?

Reply:
callback is if you're using client.setOptions in the traditional node callback way, where once it completes the callback is used to move on to the next operation. In the Node Promises syntax you would not need a callback, and just call await on the setOptions: `await client.setOptions(options);`

Reply:
ok got it, as long as customers understand the use here--- as a javascript developer seems pretty straightforward